server/src/main/java/org/apache/uniffle/server/ShuffleServer.java

```diff
@@ -328,20 +328,17 @@ public ServerStatus getServerStatus() {
     return serverStatus.get();
   }
 
-  public void setServerStatus(ServerStatus serverStatus) {
-    this.serverStatus.set(serverStatus);
-  }
-
   public synchronized void decommission() {
-    if (isDecommissioning()) {
+    if (isDecommissioning() || isDecommissioned()) {
       LOG.info("Shuffle Server is decommissioning. Nothing needs to be done.");
       return;
     }
-    if (!ServerStatus.ACTIVE.equals(serverStatus.get())) {
```

Comment:
State could change between `serverStatus.get()` and `serverStatus.set(ServerStatus.DECOMMISSIONING)`.